modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostGISDialect.java

```diff
@@ -1491,7 +1491,8 @@ public Version getVersion(Connection conn) throws SQLException {
             ResultSet rs = null;
             try {
                 st = conn.createStatement();
-                rs = st.executeQuery("select PostGIS_Lib_Version()");
+                //rs = st.executeQuery("select PostGIS_Lib_Version()");// If the PostGIS extension is in other schema (not public schema), an exception is thrown. 
+                rs = st.executeQuery("SHOW server_version");
```

Comment:
this is wrong, it does not provide the version of the postgis extension that is installed in the schema

Reply:
If you have installed postgis somewhere other than in public (why?) it is your responsibility to make sure the search path includes that location so that the postgis functions can be found. Otherwise not finding the version will be the least of your issues.